### PR TITLE
Adds node color thresholds

### PR DESCRIPTION
**Pull request by jkafader-esnet**

### CODE REVIEW

File: src/MapPanel.tsx [hunk 1]
```diff
@@ -301,8 +303,8 @@ export class MapPanel extends Component<Props> {
         ]).then((topologyData) => {
           for(let i=0; i<LAYER_LIMIT; i++){
             try {
-              let parsedData = parseData(data, topologyData[i], colors[i], fields[i], i);
-              topology[i] = parsedData[3];
+              // @ts-ignore
```
Comment (jkafader-esnet) [inline src/MapPanel.tsx:306]: ts-ignore here to let JS do its thing -- this is a weird string/object/nully set of things that can't be typed easily

File: src/components/lib/dataParser.js [hunk 2]
```diff
@@ -170,6 +189,12 @@ export function parseData(data, mapData, colors, fields, layer) {
       } else {
         infOut.push({ name: row[dstKey], value: row[outboundKey] });
       }
+
+
```
Comment (jkafader-esnet) [inline src/components/lib/dataParser.js:193]: duck typing to find out of this row relates to nodes

File: src/module.ts [hunk 3]
```diff
@@ -42,22 +42,6 @@ function checkBools(settings: object) {
   };
 }
 
-
```
Comment (jkafader-esnet) [inline src/module.ts:45]: function is now unused

File: src/module.ts [hunk 4]
```diff
@@ -42,22 +42,6 @@ function checkBools(settings: object) {
...
-        const name = getFieldDisplayName(field, frame, context.data);
-        const value = field.name;
-        options.push({ value, label: (!!frameName ? '[' + frameName + '] ' : '') + name });
-      }
-    }
-  }
-  return Promise.resolve(options);
-}
-
 async function buildChoicesWithSuggestions(context: FieldOverrideContext) {
```
Comment (jkafader-esnet) [inline src/module.ts:45]: These changes comprise a further reorganization of the options for each layer (these will be even more important with "unlimited layers" later) -- I approved these with Katrina in person

File: src/types.ts [hunk 5]
```diff
@@ -47,10 +50,9 @@ export interface MapOptions {
     center?: {
       lat: number;
       lng: number;
-      //zoom?: number;
     }
+    zoom?: number;
```
Comment (jkafader-esnet) [inline src/types.ts:54]: Correct this: Omission

**Review by jkafader-esnet — COMMENTED** [review]
Commentary for Dan

File: src/components/lib/dataParser.js [hunk 6]
```diff
@@ -243,22 +268,27 @@ export function parseData(data, mapData, colors, fields, layer) {
...
+    let nodeMatch = nodeValues.find((d) => d.name == node.name);
+    let nodeMatchValue = nodeMatch ? nodeMatch.value : 0;
+    let output = thresholdLookup(colors.nodeThresholds, nodeMatchValue);
+    if(output){
+      node.color = output;
     }
   });
 
   //take this out later
   mapJson.aTest = 0;
```
Comment (daldoyle) [inline src/components/lib/dataParser.js:290]: Was this to see if I actually looked at the MR? :)

File: src/module.ts [hunk 7]
```diff
@@ -586,6 +577,9 @@ for(let i=0; i<LAYER_LIMIT; i++){
     if(options[option].hasOwnProperty("description")){
       options[option].description = options[option].description.replace("${i+1}", (i+1).toString());
     }
+    if(options[option].hasOwnProperty("category")){
+      options[option].category = options[option].category.replace("${i+1}", (i+1).toString());
+    }
```
Comment (daldoyle) [inline src/module.ts:582]: These 3 iterations feel like they could be merged into a single function.

for var thing in ["description", "category", "defaultVault"]
   var val = options[option][thing];
   if val == undefined || not typeof(val) === 'string'
       continue
   
    options[option][thing] = val.replace("${i+1}", (i+1).toString())

**Review by daldoyle — COMMENTED** [review]
Looking good for the most part, tried not to nitpick anything. I suggested one area to condense code and make it more future proof, plus one possible debugging left in.

**Pull request merged**